src/main/java/org/sagebionetworks/bridge/file/InMemoryFileHelper.java

```diff
@@ -77,6 +77,19 @@ public void deleteDir(File dir) {
         dirSet.remove(dirPath);
     }
 
+    // To make this helper accurate, we delete the directory, then we iterate over all files and dirs and delete them
+    // if they are in this directory (have this directory + '/' as a prefix).
+    @Override
+    public void deleteDirRecursively(File dir) throws IOException {
+        // Delete directory.
+        deleteDir(dir);
+
+        // Delete all directories and files with this as a prefix.
+        String dirPrefix = dir.getAbsolutePath() + '/';
+        dirSet.removeIf(otherDir -> otherDir.startsWith(dirPrefix));
+        fileMap.entrySet().removeIf(entry -> entry.getKey().startsWith(dirPrefix));
```

Comment:
I'm surprised you can delete the directory before it's contents

Reply:
For test utils, it's all just sets and maps.